docs/sections/user_guide/cli/tools/config.rst

```diff
@@ -284,229 +278,224 @@ and an additional supplemental YAML file ``values2.yaml`` with the following con
 
-* To read the config from ``stdin`` and realize to ``stdout``:
+  The format can be explicitly specified  (``config.txt`` is a copy of ``config.yaml``):
 
   .. code-block:: text
 
-     $ cat config.yaml | uw config realize --input-format yaml --output-format yaml values1.yaml
+     $ cat config.yaml | uw config realize --update-file update.yaml --output-format yaml --input-format yaml
      values:
        date: 20240105
        empty: null
        greeting: Good Night
-       message: 'Good Night Moon Good Night Moon '
+       message: Good Night Moon Good Night Moon
        recipient: Moon
        repeat: 2
 
-* To read the config from ``stdin`` and realize a subsection to ``stdout`` in a different format:
+* This example demonstrates: 1. Reading a config from ``stdin``, 2. Extracting a specific subsection with the ``--output-block`` option, and 3. Writing the output in a different format:
 
   .. code-block:: text
 
-     $ cat config.yaml | uw config realize --input-format yaml --output-format sh --output-block values values1.yaml
+     $ cat config.yaml | uw config realize --input-format yaml --update-file update.yaml --output-block values --output-format sh
      date=20240105
      empty=None
      greeting='Good Night'
-     message='Good Night Moon Good Night Moon '
+     message='Good Night Moon Good Night Moon'
      recipient=Moon
      repeat=2
 
-* If the config file has an unrecognized (or no) extension, ``uw`` will not know how to parse its contents:
+.. note:: Combining configs with incompatible depths is not supported. ``ini`` and ``nml`` configs are depth-2, as they organize their key-value pairs (one level) under top-level sections or namelists (a second level). ``sh`` configs are depth-1, and ``yaml`` configs have arbitrary depth.
 
-  .. code-block:: text
+   For example, when attempting to generate a ``sh`` config from the original depth-2 ``config.yaml``:
 
-     $ uw config realize --input-file config.txt --output-format yaml values1.yaml
-     Cannot deduce format of 'config.txt' from unknown extension 'txt'
+   .. code-block:: text
 
-  In this case, the format can be explicitly specified  (``config.txt`` is a copy of ``config.yaml``):
+      $ uw config realize --input-file config.yaml --output-format sh
+      [2024-05-20T19:17:02]    ERROR Cannot realize depth-2 config to type-'sh' config
+
+* It is possible to provide update values, rather than the input config, on ``stdin``. Usage rules are as follows:
+
+  * Only if either ``--update-file`` or ``--update-config`` are specified will ``uw`` attempt to read and apply update values to the input config.
+  * If ``--update-file`` is provided with an unrecognized (or no) extension, or if the update values are provided on ``stdin``, ``--update-format`` must be used to specify the correct format.
+  * At least one of the input config and the update config must be provided via file: They cannot be streamed from ``stdin`` simultaneously.
```

Comment:
```suggestion
  * At least one of the input config or the update config must be provided via file: They cannot be streamed from ``stdin`` simultaneously.
```

Reply:
@NaureenBharwaniNOAA What's your thinking on this and/or (no pun intended) do you have a reference for this? I thought about this when I wrote it and landed on "and" to avoid suggesting that this is a mutually exclusive choice. It's ok for either one to be true, but also for both of them to be true. Normal usage of "or" for English speakers tends to assume mutual exclusion, for example, if I say "We have two pieces of candy left. Do you want the caramel or the peppermint?" I don't expect you to say "Yes" and eat them both, though _mathematically_ this would be a correct interpretation of "or". The accepted answer in [this StackExchange topic](https://patents.stackexchange.com/questions/20738/should-i-use-and-or-or-for-at-least-one-of-structure-when-only-two-items-i) reflects my thinking, but I'm having trouble finding anything authoritative about this. Google Docs doesn't flag either construction is grammatically incorrect. I'd be happy to take a tie-breaker vote from another reviewer...

Reply:
My thinking was with either in front of the sentence to me it implied only one or the other. But if it's the case that 1. both are accepted or 2. either one can stand alone, this changes my opinion of the sentence.

Reply:
I think the English "or" is relatively ambiguous outside of further context. To counter the candy example, if a waiter were to ask at the end of a meal "Would you like coffee or dessert?" I know it's not exclusive -- I could (and will) get both. Neither the "and" nor the "or" are carrying much weight in this case: it's the "At least one of the" at the beginning of the sentence that conveys the meaning. I think it is clear enough with "and", but could see how someone could interpret it as "at least one of _each_". That's not to say they are _correct_ to interpret it that way, but, particularly for non-natives, the grammatically correct form is not always the clearest.

Reply:
If the two options were _categories_ instead of _elements_, I think the "and" could be problematic. For example TIL that a US Circuit Court [ruled](https://www.bipc.com/ptab-establishes-plain-meaning-of-at-least-one-of-a-and-b) that "at least one of A and B", where A and B are categories, requires at least two elements -- at least one each from each of the categories; whereas "at least one of A or B" could be just a single element from either category. Lawyers!

There's probably no right answer to this given that humans have to read and interpret the words. I'll probably leave it as-is unless someone else takes a strong position.

Reply:
I was going to ask what Circuit Court the UW Team would fall under, but actually, the Federal Circuit would be a logical choice if we were interested in patenting `uwtools`... :D On a more serious note, why not reword the sentence to make the meaning clear instead of hoping that readers agree with the Federal Circuit? ;) For example, "The input config, the update config, or both must be provided via file; they cannot be streamed from ``stdin`` simultaneously." (Note that that colon should be a semicolon followed by lowercase or else a period followed by new sentence.)

Reply:
@gspetro-NOAA Excellent alternative! I will use it. As much as I'd like to establish a precedent and not have to avoid this construction forever, clarity wins the day.

I'm not sure about the punctuation and capitalization, though. Doesn't it depend on the style one follows? I was initiated into AP Style*, where the text following a colon is capitalized if it's a complete sentence (✔️), and a colon is used to clarify or complete the preceding statement -- "to introduce an explanatory statement" (✔️ IMO because the text after the colon explains _why_ the text before it is a rule). But I think maybe you have an MLA background; good stuff if so, but old habits etc.

* I'm an AP apostate because I believe in the Oxford Comma.

Reply:
Ha! I am a fellow apostate and have established the Oxford comma as one of the program's standard exceptions to AP Style (https://confluence.epic.oarcloud.noaa.gov/x/C4C2AQ). That said, I believe that AP Style's rule is to use the Oxford comma only in technical/scientific documents, so I just generalized that rule, since (a) the Oxford comma is just better (aka clearer) ;) and (b) much of what we do is technical anyway. 
As to the colon, the program doesn't have a specific rule (although I could make one... ;) ). We are loosely based on AP Style, but that is difficult to adhere to strictly without actually being given a copy of the AP Style Guide. It looks like technically, what you're doing may be allowed in AP Style, but without the style guide, the details are a little fuzzy. A colon can be used when followed by an illustration or elaboration--usually (but not necessarily?) in the form of a list. I typically see a colon used only when a list of examples follow. Technically, examples are a type of explanation/elaboration, so this seems to be a bit of a square/rectangle problem. A colon can be used with an explanatory statement that's a list, but can it be used with a broader array of explanatory statements? Possibly. But how broad an array? AP Style does say to use colons sparingly, and it seems like a period would suffice here. The idea already appears as a single bullet point, which itself suggests that the ideas are connected (otherwise they would be separate bullets). Taken to the extreme, whole paragraphs could sometimes pop in a bunch of colons since usually, ideas build on each other and continue to explain previous statements. In short, this seems like a case where you _can_ justify using a colon, but I would still air on the side of a period.